Require fields for non-nullable properties.

jmcarp opened the pull request:
This is helpful for the common use case of creating a new record from
data: if required fields are missing, we should throw an error. For
partial updates, users would pass `partial=True` to the `ModelSchema`
constructor as usual.

jmcarp committed:
Require fields for non-nullable properties.

This is helpful for the common use case of creating a new record from
data: if required fields are missing, we should throw an error. For
partial updates, users would pass `partial=True` to the `ModelSchema`
constructor as usual.

jmcarp committed:
Pass `partial` as needed in tests.

Also temporarily install marshmallow from dev--didn't realize partial
was such a new addition.

sloria commented:
I'm going to cut the marshmallow 2.2.0 release before merging this, so we don't have to change .travis.yml

sloria commented:
Though, I suppose the minimum supported marshmallow version would need to be bumped up to 2.2.0

sloria commented:
Actually, this doesn't technically break backwards compat with marshmallow 2.0.0 (`partial` is only used in the tests). Can you add a conditional for the assertions that use `partial`?

sloria commented:
Btw, marshmallow 2.2.0 is released, so you should be able to do 

``` python
marshmallow_version_info = tuple(map(int, marshmallow.__version__.split('.')))
if marshmallow_version_info >= (2, 2, 0):
    #...
```

jmcarp committed:
Deserialize complete data in tests.

Avoid the need to pass `partial` and inspect version number in the
tests: the tests are responsible for checking deserialization behavior,
not required fields or partial loading.

jmcarp commented:
IMO it's simpler just to deserialize complete objects in the tests, so we don't have to worry about `partial` marshmallow version. Patch updated.
